components/form/builder/src/index.js

```diff
@@ -0,0 +1,279 @@
+  constructor({config, submitText}) {
+    super()
+    this._config = config
+    this._formFields = Object.keys(this._config)
+    this._initialState = this._getDefaultInitialState(this._formFields)
+    this._submitText = submitText
+
+    this.state = {...this._initialState, showErrors: false}
+  }
+
+  /* Requested the list of vehicle brands by default at the start of the PTA */
+  componentDidMount = async () => {
+    const items = await this.props.onLoad()
+    const initialState = this._getDefaultInitialState(this._formFields)
+    this.setState({...initialState, ...items})
+  }
+
+  /* Get initial form state */
+  _getDefaultInitialState = fields =>
+    fields.reduce((state, key) => {
+      state[key] = INITIAL_LIST_ITEM_STATE
+      return state
+    }, {})
+
+  _getFormValues = (fields, currentField, currentValuedId) =>
+    fields.reduce((state, field) => {
+      const selectedFieldId = this._getSelectedFieldId(this.state[field])
+      state[field] = currentField === field ? currentValuedId : selectedFieldId
+      return state
+    }, {})
+
+  /* Must implement the method of calling the UseCase, passing the field name and the parameters to it */
+  _getFieldItems = async (field, currentField, currentValueId) => {
+    const params = this._getSelectedFormValues(
+      field,
+      currentField,
+      currentValueId
+    )
+    const items = await this.props.onChange({field, params})
+    return items
+  }
+
+  /* From a field name, returns the previous values to that field of the form, to make the request to UseCase through the generated parameters. */
+  _getSelectedFormValues = (nextField, currentField, currentValuedId) => {
+    const maxIndex = this._formFields.indexOf(nextField)
+    const selectedFields = this._formFields.filter(
+      (_, index) => index < maxIndex
+    )
+    const selectedFormValues = this._getFormValues(
+      selectedFields,
+      currentField,
+      currentValuedId
+    )
+    return selectedFormValues
+  }
+
+  /* Function responsible to clear the data from the fields subsequent to the current selection, except if isPersistent */
+  _clearForwardFields = field => {
+    const maxIndex = this._formFields.indexOf(field)
+    const fieldsToClear = this._formFields.filter(
+      (field, index) => index > maxIndex && !this._config[field].persists
+    )
+    const clearedFields = this._getDefaultInitialState(fieldsToClear)
+    return clearedFields
+  }
+
+  /* Get the selected field name from a list */
+  _getSelectedFieldName = list => {
+    const selectedItem = list.find(({selected}) => selected)
+    return selectedItem && selectedItem.name
+  }
+
+  /* Get the selected field */
+  _getSelectedFieldId = field => {
+    const isArray = Array.isArray(field)
+    const selectedItem = isArray && field.find(({selected}) => selected)
+    return selectedItem && selectedItem.id
+  }
+
+  /* Set the selected field and returns the modified object */
+  _setSelectedField = (field, value) =>
```

Comment:
I don't understand this method, is it setting something? Maybe it's something more like `_getNewStateWithSelectedField` or just `_getNewState`, isn't?

Reply:
You're right, thx!

Reply:
DONE!